controllers/machine_controller.go

```diff
@@ -183,9 +185,18 @@ func (r *MachineReconciler) reconcileDelete(ctx context.Context, cluster *cluste
 		}
 	} else {
 		klog.Infof("Deleting node %q for machine %q", m.Status.NodeRef.Name, m.Name)
-		if err := r.deleteNode(ctx, cluster, m.Status.NodeRef.Name); err != nil && !apierrors.IsNotFound(err) {
-			klog.Errorf("Error deleting node %q for machine %q: %v", m.Status.NodeRef.Name, m.Name, err)
-			return ctrl.Result{}, err
+
+		var deleteNodeErr error
+		err = wait.PollImmediate(2*time.Second, 10*time.Second, func() (bool, error) {
+			if deleteNodeErr = r.deleteNode(ctx, cluster, m.Status.NodeRef.Name); err != nil && !apierrors.IsNotFound(err) {
```

Comment:
you are assigning to `deleteNodeErr` and then testing against `err`

Reply:
thanks! fixed